aten/src/ATen/core/function_schema.h

```diff
@@ -78,6 +95,38 @@ struct Argument {
     return Argument(name_, new_type, N_, default_value_, kwarg_only_, alias_info_);
   }
 
+  // this function check whether this Argument is backward compatible with
+  // the old_argument. we intent to be conservative, if necessary, we may
+  // relax the checks in future. we consider the following the situations are
+  // backward compatible:
+  //   1) two arguments are equal
+  //   2) old_argument's type is T, and this Argument's type is Optional[T]
```

Comment:
This is overly specific. Instead, it should be using the isSubtype property on on types, which already handles the optional case and will future proof it for further cases. Follow the subtyping rule for functions (https://en.wikipedia.org/wiki/Subtyping#Function_types)

Reply:
agree, updated.